Dear Editor,

Please find attached our manuscript, *Gene duplications potentiate adaptive evolution of complex traits*, for your consideration.

**What is the scientific question you are addressing?**

How does gene duplication influence the evolution of biological complexity?
Specifically, we examine the role of adative processes (e.g., neo-functionalization, dose effects) in shaping gene duplication's impact on the measures of biological complexity.

**What is the key finding that answers this question?**

We find that gene duplication promotes *de novo* evolution of complex phentoypic traits via neo-functionalization.
This effect is limited to complex traits, with adaptive evolution of simpler traits exhibiting no benefit beyond the effect of increased genome length alone.
<!-- We find the adaptive benefit of gene duplication to be strongest in enabling adaptive evolution of complex, rather than simple, traits. -->
Through step-by-step analysis of lineage histories, we trace an outsized fraction of coding sites for novel complex traits back to duplicated regions; similarly, this effect is specific to complex traits, not being observed for simple traits.

**What is the nature of the evidence you provide in support of your conclusion?**

We report *in silico* evolution experiments comparing the adaptive evolution of complex traits both with and without of duplicative mutation processes, in conjunction with lineage-based analyses examining the genomic origins of evolved traits.
For this purpose, we used the Avida digital evolution platform, which enables self-replicating computer programs to be used as a model organism for evolution experiments.
To study phenotypic complexity, a framework for adaptive traits within Avida exeriments drawn from Lenski et al. (2003).
This framework defines a set of available metabolic functions with varying complexity based on the number of logical operations necessary to carry them out.

While careful consideration must be taken in generalizing results from Avida to biological organisms, in the context of our work, this model offers unique capabilities for detailed, rigorously controlled treatments with populations undergoing full Darwinian evolution.
Our computational medium enabled further work testing partial ablations of gene duplication mechanisms to separate the impact of propagating genetic information from side effects in increased genome length and increased effective mutation rate.
We also take advantage of detailed lineage records to examine exact mutational histories of coding sites for phenotypic traits to discern the role of gene duplication.

> Lenski, R. E., Ofria, C., Pennock, R. T., & Adami, C. (2003). The evolutionary origin of complex features. Nature, 423(6936), 139–144. https://doi.org/10.1038/nature01568

**What are the three most recently published articles that are relevant to this question?**

> **Ljiljana Mihajlovic et al.** (2025) A direct experimental test of Ohno’s hypothesis
eLife 13:RP97216.
--- Our finding that gene duplication can boost adaptation contrasts notably with this work, comparing rate of improvement for coGFP floursescence strength between single- vs. dual-copy populations under directed evolution conditions in an *E. coli* model

> **Tong, K. et al.** "Genome duplication in a long-term multicellularity evolution experiment." *Nature* (Mar 2025) --- Experimental evolution in *in vivo* model system, demonstrating adaptive role of dose effects.

> **DuBose and de Roode**, The link between gene duplication and divergent patterns of gene expression across a complex life cycle, Evolution Letters (October 2024) --- Establishes evidence for adaptive neo-functionalization of duplicated genes in monarch butterflies vis-a-vis shifts in expression patterns between lifecycle stages.

**What significance do your results have for the field?**

Our study delivers rare, direct‐experimental evidence that gene duplication can contribute substantially to biological complexity via adaptive innovation in complex traits.
By isolating duplicative mutational mechanisms in Avida and tracing detailed mutational steps along complete lineages, we advance the discussion to a causal demonstration that (1) gene duplications facilitate *de novo* origin of adaptive traits, (2) this effect is pronounced for complex traits, and (3) genetic information content propagated in duplications contributes to this advantage.
These findings contrast notably with framings hypothesizing sufficiency of non-adaptive explanations for biological complexity.
<!-- As such, this work provides a cornerstone complementary to comparative approaches in natural history. -->


**What significance do your results have for the broader community (of biologists and/or the public)?**

From a science communication perspective, lines of research working with ``evolution in action'' provide exciting opportunity to broaden public perceptions of evolutionary processes as an ongoing story rather than a historical record.
In concert with the broader experimental evolution community, experimenting with digital evolution models is notable for its low barrier to entry through web-based applications such as Avida-ED.

Our research also pertains to application-oriented work engineering evolution as an optimization process, both *in vivo* and *in silico*.
To these ends, we contribute to the body of evidence highlighting duplicative processes as key for adaptive evolution and, in particular, our research defines problem contexts where gene duplication processes may be most impactful (e.g., complex problems with potential to assemble functional building blocks).

**What other novel findings do you present?**

- Ablation experiments demonstrate that genetic information propagated by slip duplications, specifically, contributes to subsequent adaptive innovation.
- We show that slip duplication accelerates the accumulation of vestigial coding sites in the genome, beyond the impact of increased genome size alone.
- We see evidence that contingent processes is necessary to adaptive effects of gene duplications --- by preventing loss of duplicated material from selection against mutational load associated with genome size.

**Is there additional information that we should take into account?**

Our project builds on an impactful body of experimental work using Avida to study aspects of the evolution of complexity, notably including two works appearing in PLOS Biology --- Zaman et al. 2014 (doi:[10.1371/journal.pbio.1002023](https://doi.org/10.1371/journal.pbio.1002023)) and Goldsby et al. 2014 (doi:[10.1371/journal.pbio.1001858](https://doi.org/10.1371/journal.pbio.1001858)).
Note also that an earlier version of this work appeared at the 2017 Artificial Life conference (doi:[10.1162/isal_a_045](https://doi.org/10.1162/isal_a_045)).
We look forward to further contributing to conversation around the evolution of complexity.

Thank you for your consideration,
The Authors
