\section{Introduction} \label{sec:introduction}

Genetic surveys and evolution experiments have made available an increasingly detailed picture of how duplicative processes shape genetic information.
In experimental yeast systems with harsh conditions, adaptations resulting from changes in chromosome count (i.e., aneuploidy) confer superior fitness \citep{Pavelka:2010}.
Similarly, cancer cells are commonly observed in polyaneuploid states, and the drastic genetic changes from duplications and deletions could play an important role in creating resistance mechanisms \citep{polyaneuploidCancer}.
Concrete phenotypic effects have been directly attributed to changes in copy count, such as the short leg length characteristic of dachshunds arising from an additional copy of the FGF4 gene \citep{dachshundGeneCopyNum}.

Duplications in genetic material range from repeats of gene fragments to whole genome duplication.
By providing new genetic material, these events are thought to introduce potential for further evolutionary modification.
As such, duplication is understood to be an essential driver for genetic variation \citep{Zhang:2003fw,Crow:2006role,Magadum:2013wu}.
Indeed, gene duplication has been shown to promote adaptive evolution in both biological and simulation models \citep{Hu:2010ea}.
% [x] AML (2025): I shortened the above definition of evolvability for succintness (folks should make sure they still agree with it). Even better, if we could streamline the sentence to not have a definition interrupt the flow.
% MAM Done; I've been scraping out most discussion of evolvability in favor of just discussion "promotion of adaptive evolution"

A striking example of gene duplication leading to new adaptations
comes from the Long-Term Evolution Experiment in \textit{Escherichia coli}, where a duplication in one population broadened expression of key metabolic machinery to previously inhibitive conditions \citep{blount_genomic_2012}.
As a result, the population was able to metabolize citrate, a previously inaccessible carbon resource, resulting in a 7-fold increase in population size.

In addition to such anecdotal case studies, large-scale genomic analyses have discovered cases where a strikingly high fraction of the genes in an organism show evidence of having arisen from gene duplications \citep{teichmann_structural_1998,Teichmann:2004cz}.
Comparative studies have associated duplication events early in natural history with increases in genetic robustness and evolutionary innovation \citep{wagner_gene_2008}.

The prominent role of gene duplications in biological evolution has inspired incorporation of analogous mechanisms in evolution-inspired optimization algorithms \citep{Ryan:1998gm,Sawai:1999genetic,Sawai:2000comparative,Schmitt:2005bc}.
Notably, in genetic programming, gene duplication and deletion operators have been shown to increase program evolvability and yield simpler evolved solutions \citep{Koza:1995fr}.
In work evolving neural controllers for robots, enabling module duplication was found to increase functional specialization in network modules \citep{Calabretta:1998vh,Calabretta:2000tl}.

Given the evidence that gene duplication can facilitate adaptive evolution in both computational and natural systems, we use a digital evolution approach to explore \textit{why}.
That is, what mechanistic aspects of gene duplications promote adaptive evolution?

One question is how fidelity of duplicated material influences subsequent evolutionary outcomes.
Exact duplications can result in functionally redundant genes that can increase the mutational robustness of a genotype \citep{Crow:2006role} or allow the organism to produce additional gene product \citep{Zhang:2003fw}.
If a highly constrained genetic sequence is duplicated, one copy can potentially mutate more freely and produce new functionality (i.e., neofunctionalization) \citep{Zhang:2003fw,Wagner:2003fk}.
Alternatively, subfunctionalization may occur where both gene copies diverge from the ancestral gene state, specializing in complementary aspects of the ancestral gene's functionality \citep{Zhang:2003fw}.

It is also possible that ``side effects'' of gene duplication may contribute to adaptive evolution, such as effects in increasing effective mutation rate, localized clustering of sequence changes, and increases in genome length.
Due to their inherent co-occurrence, it is not obvious how to disentangle these aspects of gene duplication in field or laboratory studies.

Interest in duplication of genetic material as an evolutionary catalyst dates well before modern understanding of how genetic information is stored and processed in biological organisms \citep{Metz:chromosomeDuplication1947}.
Today, digital systems provide an opportunity to model sophisticated evolutionary processes in a framework where they can be observed in complete detail.

Using the Avida Digital Evolution Platform \citep{ofria2004avida}, we implemented a series of mutation operators to systematically isolate aspects of gene duplication and tease apart which factors promote evolvability.
Specifically, we use gene duplication mutation operators in Avida that resemble replication slippage \citep{bzymek_instability_2001} (\textit{slip mutations}) and allow for gene duplications or deletions at any scale.
When slip mutations were present, we found that the populations evolved complex tasks significantly faster.
Moreover, by analyzing the dominant lineage of each population, we observed that complex tasks, when first evolved, were signficantly more likely to depend on instructions within duplicated regions of the genome.
Together, these results strongly suggest that gene duplication plays a pivotal role in the evolution of traits with multiple components, providing insight into the origin of intricate natural systems.

% \subsection{Major Results}

% We found local slip mutations that duplicate intact regions to be the most effective configuration of gene duplication in facilitating evolution on the Logic-9 task set within the Avida platform.
% In particular, we found that --- compared to control experiments with long genome sizes --- gene duplication uniquely promoted the evolution of complex adaptive traits.
% We further found that the raw material created by slip duplication plays a potentiating role in the evolution of complex traits.
% Specifically, we identified that slip-duplicated regions are significantly more likely to serve as coding sites for new traits when they first appear.
% Consistent with expectations under neofunctionalization theory, however, we did not observe potentiation effects of slip duplication on the evolution of simple traits, only traits that were facilitated by multiple building block components.

% % [x] @CAO: Since I'm making a lot of changes, I'll start duplicating and commenting out the original text.  I hear that such duplications can accelerate the adaptive process.  ;-)
% Finally, we assessed the consequences of slip duplication on genome architecture.
% One hypothesis is that gene duplication would promote genetic brittleness by increasing genome length and accelerating growth in contingent complexity, as newly redundant genetic material specializes over time.
% % [x] @CAO Do we need to define "genome complexity" before we use it here?
% % @MAM Perhaps we can say something a little more specific instead of "genome complexity"
% Contrary to this possibility, we found that active coding sites grew as a rate similar to control experiments.
% However, we observed that slip duplications produced a significant increase in the accumulation of coding material, when active and vestigial sites are considered together.
% To understand this phenomenon, we tested the immediate effects of slip duplication on genome brittleness.
% We found that, on average, fitness-neutral slip duplications decreased, rather than increased, the number of information-baring sites in a genome.
% These results align with our observed increase in vestigial coding material in genomes.
% These brittleness-reducing effects appear to be counteracted by other factors, resulting in a similar overall trajectory of genome information content between the slip-duplication and control treatments.
