\section{Methods} \label{sec:methods}

\subsection{The Avida Digital Evolution Platform}

We conducted experiments using Avida, which instantiates an evolutionary process among self-replicating computer programs acting as ``digital organisms.''
This agent-based modeling approach allows research to empirically test hypotheses about evolution that would be difficult or impossible to test in natural systems \citep{Ofria:2009avida}.

\input{fig/vhardware.tex}

Digital Organisms in Avida compete to acquire resources and replicate within limited space in a finite population.
Each organism is defined by a linear sequence of program instructions (\textit{i.e.} its genotype) and a set of virtual hardware. The instruction set in Avida is Turing Complete and syntactically robust -- any ordering of instructions is syntactically valid, though not always useful. The instruction set includes operations for basic computations, execution flow control, input and output, and self-replication. An organism's virtual hardware (depicted in Figure \ref{fig:vhardware}) consists of components such as a central processing unit (CPU), registers for computation, buffers for input and output, and memory stacks.

Organisms in Avida replicate asexually by allocating memory for their offspring, copying their genome instruction-by-instruction, and then dividing. However, copy and divide operations have imperfect fidelity and can result in mutated offspring. Organisms can influence their replication speed by improving their metabolic rate. An organism's metabolic rate determines the speed at which it executes instructions in its genome; a higher metabolic rate allows an organism to execute its genome faster, and thus, allows the organism to copy itself faster. Initially, an organism's metabolic rate is set proportional to its length so as to prevent intrinsic disadvantages from increased copy loop duration for long genomes. However, an organism's metabolic rate can be improved when it completes a particular task, such as a mathematical computation. In this way, we can differentially reward the performance of computational tasks.

When an organism finishes replicating, its offspring are placed in a random location elsewhere in the population, replacing the previous occupant of that location.
Thus, improving the efficiency of self-replication or performing rewarded computational tasks are both advantageous in the competition for space in Avida. The combination of competition and heritable variation due to imperfect copy and divide operations results in evolution by natural selection in Avida \citep{pennock2007models}.

\input{fig/slip_mut_variants.tex}

\textbf{Mutation events in Avida} come in four types: substitution, insertion, deletion, or slip.
Rates of occurrence for each were independently configured, as described next.

% https://github.com/chaynes2019/AvidaGeneDupe/blob/375f1a3018669f50d00e4ce67b977931cf42f6d9/experiments/2022-3-29-PaperDuplication/hpcc/config/avida.cfg
\textit{Copy mutations} may introduce an erroneous \textit{substitution} where a random instruction is written to the offspring genome instead of the intended instruction.
In our experiments, such substitutions occurred with a per-site probability of 0.0025.

\textit{Divide mutations} act on an organism's offspring during division.
When a divide \textit{insertion} mutation occurs, an arbitrary instruction is inserted at random into the offspring's genome, increasing the size of the offspring's genome by one.
When a divide \textit{deletion} mutation occurs, a random instruction in the offspring's genome is removed, decreasing the size of the offspring's genome by one.
In our experiments, each was allowed to occur with 5\% probability per offspring.

For the purposes of this study, we augmented Avida with a new type of divide mutation: \textit{Slip mutation}.
This type of mutation acts analogously to gene duplications and deletions.
When a slip mutation occurs, two sites in the offspring genome are randomly selected, defining the target segment for the operation.
Suppose the first site is upstream of the second.
In that case, the slip mutation results in an insertion --- as if the organism's replication machinery had slipped backward during replication and recopied a segment.
If the second site is upstream of the first, the slip mutation results in a deletion --- as if the organism's replication machinery slipped forward during replication, skipping over a genetic segment.
As implemented, slip-insertions and slip-deletions occur with equal probability; thus, absent selection, this mutational process does not introduce an inherent bias on genome length.

Across trials, we assess five variants of slip mutations: slip-duplicate, slip-scramble, slip-random, slip-NOP, and slip-scatter.
When a slip mutation results in a deletion, in all but the slip-scatter variant, the target segment is deleted. Rather than deleting the target segment, the slip-scatter mutation operator distributes a number of single-instruction deletions equal to the length of the target segment uniformly throughout the genome.
When a slip mutation results in an insertion, a number of instructions equal to the length of the target segment is inserted into the genome; however, depending on the particular slip mutation operator, which instructions are inserted and where the insertions take place may be different.
Figure \ref{fig:slip_mut_variants} provides description and examples of how each slip mutation operator handles insertions.
Where enabled, slip mutations occurred with 5\% probability per divide event.

\subsection{Experimental Design}

\input{tab/tasks.tex}

Across all experiments, we rewarded the performance of nine Boolean logic functions: NOT, NAND, OR-NOT, AND, OR, AND-NOT, NOR, XOR, and EQUALS.
Detailed description of this ``Logic-9'' task set can be found in \citet{lenski2003evolutionary} ---
however, one significant characteristic is well-defined differences in difficulty between tasks, determined by an intrinsic computational complexity measure (``NAND count'') --- given for each in Table \ref{tab:tasks}.
Identical reward was provided for each task and kept consistent throughout trials.
We measured adaptation to the Logic-9 task set as a ``phenotypic match score,'' ranging from a minimum of 0 (\textit{i.e.} the organism performs no tasks) to a maximum of 9 (\textit{i.e.} the organism performs all 9 tasks).

We ran evolutionary experiments for 20,000 Avida updates.%
\footnote{An update in Avida is equal to the amount of time it takes for the average organism to execute 30 instructions; see \citep{Ofria:2009avida} for further detail.}
This duration was sufficient to observe at least 600 generations within all trials.
To assess evolutionary history within a trial, we extracted the trial's concluding final dominant genotype and recorded its lineage ancestry.
In a postprocessing step, we applied Avida's ``analyze mode'' to identify the set of tasks performed by each ancestor.
For investigations involving the evolutionary history of individual genome sites, we used mutational metadata saved with lineage files to identify correspondences between parent and offspring.

% \input{tab/treatments.tex}

Our experiment consisted of seven treatments: two baseline treatments, with slip-duplication disabled, and five experimental treatments, for which slip-duplication was enabled.
Treatments differed only in the available mutation operators and the rates at which those operators were applied.
Each treatment was designed to tease apart an aspect of how gene duplications might promote evolvability.

% Baseline treatment
The \textbf{baseline treatment} was used as a control; we used the results from this treatment as a baseline for evolvability, with which we compared the results from all other experimental treatments. Mutation rates in the baseline treatment have been shown to facilitate both the evolution of complex boolean logic tasks, such as EQUALS, and the evolution of task regulation in Avida \citep{lenski2003evolutionary, Lalejini:2016plasticity}

% Long genome treatment
To better discern the role in facilitating evolution of larger genome size caused by gene duplication, we performed an additional treatment: the \textbf{long-genome baseline treatment}.
Genomes in this treatment operated identically to the baseline treatment above, except that they were 1,000 sites long instead of 100 sites long.
We chose this size as it was near the upper bound of the largest genome sizes that we observed in the slip-duplication treatment experiments.

% Slip-duplicate treatment
The \textbf{slip-duplicate treatment} allowed for mutation events that resulted in full code duplications via the slip-duplicate mutation operator. Duplications in this treatment preserved both the content and the structure of duplicated code. This allowed us to answer the following question: how important is it that gene duplications can exactly duplicate sequences in a genome in both content \textit{and} structure?

% Slip-scramble treatment
The \textbf{slip-scramble treatment} allowed for mutation events (via the slip-scramble operator) that resulted in code duplications where the content of the duplicated code was preserved, but the structure of the duplicated code was not preserved. This, compared with the slip-duplicate treatment, allowed us to answer the question: Is the duplication of particular instructions important, regardless of their arrangement?

% Slip-random treatment
The \textbf{slip-random treatment} allowed us to answer the following question: is it the case that gene duplications promote evolvability because they result in the insertion of large, clustered mutations, regardless of what those mutations may be? The slip-random treatment (via the slip-random mutation operator) allowed for mutation events that could insert large, contiguous clusters of random instructions; one could also think of these mutations as maximally noisy duplications where neither the content nor structure of the duplicated code is preserved.

% Slip-NOP treatment
The \textbf{slip-NOP treatment} allowed for mutations capable of inserting contiguous segments of blank `genetic tape' (via the slip-NOP mutation operator) in the form of no operation instructions. This allowed us to answer the following question: how important is it that gene duplications provide evolution with a straightforward technique for increasing genome size?

% Slip-scatter treatment
The \textbf{slip-scatter treatment} helped us tease apart whether gene duplications promote evolvability because they inflate the effective mutation rate, generating increased amounts of genetic variation. This treatment allowed for mutations that, when triggered, could insert many random instructions into random locations in a genome (via the slip-scatter mutation operator).

In all experiments, organisms were limited to a minimum genome size (\textit{i.e.} instruction sequence length) of 100\footnote{
In exploratory experiments, we found that, without enforcing a minimum genome size, slip mutations caused many lineages to quickly shrink in genome size because of inherent selection pressure for smaller genome size. Organisms became fast replicators but were then trapped on a local fitness optimum, unable to evolve to perform complex computational tasks.
}.
We limited the population size to 3600 and seeded each experiment with an ancestral genotype capable of self-replication, but no Logic-9 tasks.
We ran between 30 and 100 trials of each treatment.

\subsection{Statistical Methods}

To determine if any treatments were significant within a set, we performed a Kruskal-Wallis test, applying a Bonferroni correction (for an earlier experimental design incorporating two additional surveyed environments) to keep the experiment-wise $\alpha$ = 0.05.
For an environment in which the Kruskal-Wallis test was significant, we performed Mann-Whitney U tests for each experimental treatment against the control and applied a Bonferroni correction for the six such tests within each environment.

\subsection{Software and Data Availability} \label{sec:materials}

Supporting software and executable notebooks for this work are available via GitHub at \url{https://github.com/chaynes2019/AvidaGeneDupe/} \citep{david_m_bryson_2025_14911296}.
Simulation data is archived via the Open Science Framework at \url{https://osf.io/j5s4h/} \citep{foster2017open}.

This project benefited significantly from a large number of open-source scientific software projects \citep{2020SciPy-NMeth,harris2020array,reback2020pandas,mckinney-proc-scipy-2010,waskom2021seaborn,hunter2007matplotlib,moreno2023teeplot,r_core_team_r:_2015}.
